ParserTests/CommentsTests.cs

```diff
@@ -235,4 +461,618 @@ public void TestMixedEOLComment()
+            Assert.Equal("1", intToken1.Value);
+            Assert.Equal(0, intToken1.Position.Line);
+            Assert.Equal(0, intToken1.Position.Column);
+
+            Assert.Equal(SingleLineCommentsToken.INT, intToken2.TokenID);
+            Assert.Equal("2", intToken2.Value);
+            Assert.Equal(1, intToken2.Position.Line);
+            Assert.Equal(0, intToken2.Position.Column);
+            Assert.Equal(SingleLineCommentsToken.INT, multiLineCommentToken.TokenID);
+            Assert.Equal("2", multiLineCommentToken.Value);
+            Assert.Equal(1, multiLineCommentToken.Position.Line);
+            Assert.Equal(11, multiLineCommentToken.Position.Column);
+            Assert.Equal(SingleLineCommentsToken.DOUBLE, doubleToken.TokenID);
+            Assert.Equal("3.0", doubleToken.Value);
+            Assert.Equal(2, doubleToken.Position.Line);
+            Assert.Equal(22, doubleToken.Position.Column);
+        }
+
+        [Fact]
+        public void TestGenericSingleLineComment()
+        {
+            var lexerRes = LexerBuilder.BuildLexer(new BuildResult<ILexer<SingleLineCommentsToken>>());
+            Assert.False(lexerRes.IsError);
+            var lexer = lexerRes.Result as GenericLexer<SingleLineCommentsToken>;
+
+            var dump = lexer.ToString();
+
+            var tokens = lexer.Tokenize(@"1
+2 // single line comment
+3.0").ToList();
+
+            Assert.Equal(5, tokens.Count);
+
+            var token1 = tokens[0];
+            var token2 = tokens[1];
+            var token3 = tokens[2];
+            var token4 = tokens[3];
+
+            Assert.Equal(SingleLineCommentsToken.INT, token1.TokenID);
+            Assert.Equal("1", token1.Value);
+            Assert.Equal(0, token1.Position.Line);
+            Assert.Equal(0, token1.Position.Column);
+            Assert.Equal(SingleLineCommentsToken.INT, token2.TokenID);
+            Assert.Equal("2", token2.Value);
+            Assert.Equal(1, token2.Position.Line);
+            Assert.Equal(0, token2.Position.Column);
+            Assert.Equal(SingleLineCommentsToken.COMMENT, token3.TokenID);
+            Assert.Equal(" single line comment", token3.Value.Replace("\r", "").Replace("\n", ""));
+            Assert.Equal(1, token3.Position.Line);
+            Assert.Equal(2, token3.Position.Column);
+            Assert.Equal(SingleLineCommentsToken.DOUBLE, token4.TokenID);
+            Assert.Equal("3.0", token4.Value);
+            Assert.Equal(2, token4.Position.Line);
+            Assert.Equal(0, token4.Position.Column);
+        }
+
+        [Fact]
+        public void TestInnerMultiComment()
+        {
+            var lexerRes = LexerBuilder.BuildLexer(new BuildResult<ILexer<SingleLineCommentsToken>>());
+            Assert.False(lexerRes.IsError);
+            var lexer = lexerRes.Result as GenericLexer<SingleLineCommentsToken>;
+
+            var dump = lexer.ToString();
+
+            var code = @"1
+2 /* inner */ 3
+4
+            ";
```

Comment:
How should the parser behave here? Right now it ignores the comment as well as the INT 3 and 4, generating two tokens for the first INT (before comment) and one empty token at the end.

I would expect the comments and text to be completely ignored and the INT 3 and 4 be tokenized or to completely fail compiling with an error. The actual behavior seems erroneous for me.

Reply:
the same here as it's again the SingleLineCommentTokens lexer used.